test/test_servers.py

```diff
@@ -0,0 +1,197 @@
+import pytest
+from tabulate import tabulate
+from click.testing import CliRunner
+
+from faculty_cli.cli import cli, _server_spec
+
+from faculty.clients.server import ServerClient
+from test.fixtures import (
+    PROJECT,
+    SHARED_SERVER,
+    DEDICATED_SERVER,
+    DEDICATED_RESOURCE,
+    SHARED_RESOURCE,
+    USER_ID,
+)
+
+
+@pytest.fixture
+def mock_update_check(mocker):
+    mocker.patch("faculty_cli.update.check_for_new_release")
+
+
+@pytest.fixture
+def mock_check_credentials(mocker):
+    mocker.patch("faculty_cli.cli._check_credentials")
+
+
+@pytest.fixture
+def mock_user_id(mocker):
+    mocker.patch("faculty_cli.auth.user_id", return_value=USER_ID)
+
+
+def test_list_all_servers(
```

Comment:
The mocking of the underlying library happens at inconsistent levels here.

1. Listing projects has a mock on the `faculty_cli.cli._list_projects` helper in this codebase.
2. Listing servers mocks the 'private' `_get` method on the `ServerClient`.

This is partially because we're calling `_get` directly from the CLI code, but this will be tidier once that call is factored out into an actual method on the client.